google/resource_google_project_default_service_accounts_test.go

```diff
@@ -0,0 +1,287 @@
+					preventRaceCondition(), // Looks like the API is eventually consistent by couple seconds
+					testAccCheckGoogleProjectDefaultServiceAccountsChanges(t, project, action),
+				),
+			},
+		},
+	})
+}
+
+func TestAccResourceGoogleProjectDefaultServiceAccountsDelete(t *testing.T) {
+	t.Parallel()
+
+	org := getTestOrgFromEnv(t)
+	project := fmt.Sprintf("tf-project-%d", randInt(t))
+	billingAccount := getTestBillingAccountFromEnv(t)
+	action := "DELETE"
+	restorePolicy := "REACTIVATE"
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGoogleProjectDefaultServiceAccountsRevert(t, project, action),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckGoogleProjectDefaultServiceAccountsAdvanced(org, project, billingAccount, action, restorePolicy),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("google_project_default_service_accounts.acceptance", "id", "projects/"+project),
+					resource.TestCheckResourceAttrSet("google_project_default_service_accounts.acceptance", "project"),
+					resource.TestCheckResourceAttr("google_project_default_service_accounts.acceptance", "action", action),
+					resource.TestCheckResourceAttrSet("google_project_default_service_accounts.acceptance", "project"),
+					preventRaceCondition(), // Looks like the API is eventually consistent by couple seconds
+					testAccCheckGoogleProjectDefaultServiceAccountsChanges(t, project, action),
+				),
+			},
+		},
+	})
+}
+
+func TestAccResourceGoogleProjectDefaultServiceAccountsDeprivilege(t *testing.T) {
+	t.Parallel()
+
+	org := getTestOrgFromEnv(t)
+	project := fmt.Sprintf("tf-project-%d", randInt(t))
+	billingAccount := getTestBillingAccountFromEnv(t)
+	action := "DEPRIVILEGE"
+	restorePolicy := "REACTIVATE"
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGoogleProjectDefaultServiceAccountsRevert(t, project, action),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckGoogleProjectDefaultServiceAccountsAdvanced(org, project, billingAccount, action, restorePolicy),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("google_project_default_service_accounts.acceptance", "id", "projects/"+project),
+					resource.TestCheckResourceAttrSet("google_project_default_service_accounts.acceptance", "project"),
+					resource.TestCheckResourceAttr("google_project_default_service_accounts.acceptance", "action", action),
+					resource.TestCheckResourceAttrSet("google_project_default_service_accounts.acceptance", "project"),
+					preventRaceCondition(), // Looks like the API is eventually consistent by couple seconds
+					testAccCheckGoogleProjectDefaultServiceAccountsChanges(t, project, action),
+				),
+			},
+		},
+	})
+}
+
+func preventRaceCondition() func(s *terraform.State) error {
```

Comment:
What if instead this function were moved to provider_test.go and given a more generic name like `sleepForTest` and the ability to choose how long you sleep for? I know there's at least one other test in the provider (in provider_test.go) that has a sleep step.

Reply:
Sorry, I could not find this function. I honestly don't think we need a function for it. time.Sleep is enough.
what if we use just an inline function? 
`func(s *terraform.State) error { time.Sleep(5 * time.Second); return nil },`

Reply:
I did it... it's simple enough :D

Reply:
That's fine!

Reply:
should I update the two functions that use the sleep builtin?

Reply:
I'm not totally sure what you mean by that, but do what you think makes the most sense and I'll let you know if there's something else I'd prefer!

Reply:
I mean that I end up creating the function in the provider_test.go. Should I update the other two functions that use time.Sleep inside the provider_test.go to the function I just created? Sorry for the confusing message.
I think I shouldn't but just double-checking.

Reply:
> I'm not totally sure what you mean by that, but do what you think makes the most sense and I'll let you know if there's something else I'd prefer!

k... i will do that